clients/drcachesim/analyzer.h

```diff
@@ -115,30 +116,59 @@ template <typename RecordType, typename ReaderType> class analyzer_tmpl_t {
 protected:
     typedef scheduler_tmpl_t<RecordType, ReaderType> sched_type_t;
 
+    // Tool data for one shard.
+    struct analyzer_tool_shard_data_t {
+        analyzer_tool_shard_data_t()
+            : shard_data_(nullptr)
+        {
+        }
+        analyzer_tool_shard_data_t(analyzer_tool_shard_data_t &&src)
+        {
+            shard_data_ = src.shard_data_;
+            interval_snapshot_data_ = std::move(interval_snapshot_data_);
+        }
+
+        void *shard_data_;
+        std::queue<typename analysis_tool_tmpl_t<RecordType>::interval_state_snapshot_t *>
```

Comment:
Because the intervals are processed like a queue during merge; we always check the next oldest interval for each shard, in each iteration of the merge logic. Using a queue obviates the need to use a separate index var.

Reply:
That doesn't make sense to me: you created a collection by appending; then you walk the collection from the front.  Using a queue makes the reader think it's used in a way that needs a queue, with removal from both ends or with interleaved removal and additions: but none of that is done.

Reply:
Let's consolidate our discussion in the clients/drcachesim/analyzer.cpp:541 comment thread.